app/models/search/work_search_form.rb

```diff
@@ -98,6 +99,25 @@ def standardize_creator_queries
     @options[:query] = @options[:query].gsub('creator:', 'creators:')
   end
 
+  def standardize_language_ids
+    # Maintain backward compatibility for old work searches/filters:
+
+    # - Using language IDs in the "Language" dropdown
+    if @options[:language_id].present? && @options[:language_id].to_i != 0
+      language = Language.find_by(id: options[:language_id])
+      options[:language_id] = language.short if language.present?
+    end
+
+    # - Using language IDs in "Any field" (search) or "Search within results" (filters)
+    if @options[:query].present?
+      @options[:query] = @options[:query].gsub(/\blanguage_id\s*:\s*(\d+)/) do |m|
+        lang = Language.find_by(id: "#{$1}")
+        lang = Language.default if lang.blank?
```

Comment:
Could we remove the language search param if the language can't be found? Allow more works to be returned rather than restrict to the Archive default?

Reply:
If people use `language_id: 1` as a clause in an OR/AND search, removing it will cause a query syntax error (similar to [AO3-5104](https://otwarchive.atlassian.net/browse/AO3-5104?focusedCommentId=355062&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-355062)).